sb3_contrib/ppo_recurrent/ppo_recurrent.py

```diff
@@ -393,22 +396,19 @@ def train(self) -> None:
                 # Normalize advantage
                 advantages = rollout_data.advantages
                 if self.normalize_advantage:
-                    advantages = (advantages - advantages.mean()) / (advantages.std() + 1e-8)
+                    advantages = (advantages - advantages[mask].mean()) / (advantages[mask].std() + 1e-8)
 
                 # ratio between old and new policy, should be one at the first iteration
                 ratio = th.exp(log_prob - rollout_data.old_log_prob)
 
                 # clipped surrogate loss
                 policy_loss_1 = advantages * ratio
                 policy_loss_2 = advantages * th.clamp(ratio, 1 - clip_range, 1 + clip_range)
-                # Mask padded sequences
-                policy_loss_1 = policy_loss_1 * rollout_data.mask
-                policy_loss_2 = policy_loss_2 * rollout_data.mask
-                policy_loss = -th.min(policy_loss_1, policy_loss_2).mean()
+                policy_loss = -th.mean(th.min(policy_loss_1, policy_loss_2)[mask])
```

Comment:
mask should be applied directly to `policy_loss_1` and `policy_loss_2`, no?

Reply:
That's true, `policy_loss_1` and `policy_loss_2` are not strictly masked with zeros (due to the standardisation of the advantages and the ratios equating to ones for the masked indexes), but since we select only the valid indexes in `policy_loss` these are neglected anyway.